vcd/resource_vcd_org_vdc_test.go

```diff
@@ -332,6 +342,11 @@ resource "vcd_org_vdc" "{{.VdcName}}" {
     default  = true
   }
 
+  metadata = {
+    vdc_metadata = "VDC Metadata"
+    vdc_metadata2 = "VDC Metadata2"
```

Comment:
Would be useful to test a bigger more complex value - something at the edge of what API supports in terms of length and special characters.

Reply:
No idea, API doesn't say anything and we don't restrict anything. General approach we we left validate API as much as possible. So not sure what to test additionally.

Reply:
It can be abused with such metadata:
```
metadata = {
    vdc_metadata = "VDC Metadata"
    vdc_metadata2 = "VDC Metadata2"
    "strange-key!@#$%;," = "asd"
  }
```
And the error is quite ugly:
```
Error: error updating metadata to VDC: &errors.errorString{s:"error adding metadata: &errors.errorString{s:\"error adding metadata: [ParseErr]: error parsing error body for non-200 request: EOF (&{Status:500 Server Error StatusCode:500 Proto:HTTP/1.1 ProtoMajor:1 ProtoMinor:1 Header:map[Date:[Fri, 05 Jul 2019 08:14:44 GMT] X-Vmware-Vcloud-Ceip-Id:[946aa394-4f82-4ef7-bc04-e4054467a838] X-Vmware-Vcloud-Request-Id:[92d4f455-49f4-4b86-bd1f-88fae324da52]] Body:0xc001122380 ContentLength:-1 TransferEncoding:[] Close:true Uncompressed:false Trailer:map[] Request:0xc0014a6600 TLS:0xc00001f550})\"}"}
```
However I agree with Vaidotas - we can't doo much of a safeguard here and the API is not helpful - it gives 500 without a lot of explanations.

Reply:
only possible way validate string with regexp, but we don't know limitations

Reply:
OK, at least the error says which field is at cause: "error updating **metadata** to VDC"